src/functions.jl

```diff
@@ -0,0 +1,47 @@
+using DataFrames
+"""
+    npt_running_average(x_val, y_val, n = 5, x_name = "x", y_name = "y")
+    
+# Defines a running average function
+#####################################################################################################
+# Input:                                                                                            #
+# x_val: x-values in the data-set which will be averaged.                                           #
+# y_val: y-values in the data-set which will be averaged.                                           #
+# n: Default 5. Defines how long the running average will be.                                       #
+# x_name: Default "x". Defines the name of the x-value column in the output DataFrame.              #
+# y_name: Default "y". Defines the name of the y-value column in the output DataFrame.              #
+#                                                                                                   #
+# Output:                                                                                           #
+# df: DataFrame of the averaged y-values and respective x values with given or default column-names #
+#####################################################################################################
+"""
+function npt_running_average(x_val, y_val, n = 5, x_name = "x", y_name = "y")
+    y_temp = []                               # Defines a temporary y-value array where the averaged values will be inserted
```

Comment:
Comments to a code line should immediately *precede* the line of code. The exception is for very short comments. Also, the line length should not exceed 92, so if comments are longer than that, split it over multiple lines. You can make VSCode show a vertical line at 92 character width.